### PR TITLE
check compound key element count in riak_kv_pb_timeseries:make_ts_key/2

### PR DESCRIPTION
**Pull request by hmmr**
Instead of a badarg trace from lists:zip, get and delete will now return
# rpberrresp{11, "Need, Got"} unless cmpound key has the right number of

elements.

This should fix https://github.com/basho/riak_kv/issues/1257.

**Update:** Include a commit that moves all error codes to 1000 and above. See https://bashoeng.atlassian.net/browse/RIAK-1437 for details.

### CODE REVIEW

File: src/riak_kv_pb_timeseries.erl [hunk 1]
```diff
@@ -64,6 +64,7 @@
 -define(E_MISSING_TS_MODULE, 8).
 -define(E_DELETE,   9).
 -define(E_GET,     10).
+-define(E_BAD_KEY_LENGTH, 11).
```
Comment (lukebakken) [inline src/riak_kv_pb_timeseries.erl]: I suggest that we define TS error codes to be in a certain range that is outside of 1-1000. There will be work done soon to improve error reporting to Riak clients and each application within Riak will use its own set of codes. See RIAK-1437 in JIRA.

Comment (hmmr) [inline src/riak_kv_pb_timeseries.erl]: I'm all for it (that's what I'm kind of suggesting in the comment above these defines). However, working within the scope of this PR, let's leave it as is for now. Whoever undertakes range assignment, will do so globally and in a separate PR.

Comment (lukebakken) [inline src/riak_kv_pb_timeseries.erl]: The thing is, however, is that from the release of Riak TS onward, these codes will have meaning. If the codes change after that point, documentation will have to change, Client Services' understanding of the codes will have to change, etc - there will be a ripple effect.

I think it's a safe bet to choose something like 1001-1500 as "reserved" for timeseries. Best case, the codes will never change in the future and everyone will be happy. Worst case, they will, in which case we would be in the same boat as if they stay 1-11.

What we should do is add a note or a sub-ticket to RIAK-1437 saying that 1001-1500 have been reserved for Timeseries.

File: src/riak_kv_pb_timeseries.erl [hunk 2]
```diff
@@ -355,26 +375,32 @@ put_data(Data, Table, Mod) ->
...
+                   || {K, _} <- VoidRecord, lists:member(K, KeyFields)]),
+
+            %% 2. make the PK and LK
+            PK = eleveldb_ts:encode_key(
+                   riak_ql_ddl:get_partition_key(DDL, BareValues)),
+            LK = eleveldb_ts:encode_key(
+                   riak_ql_ddl:get_local_key(DDL, BareValues)),
+            {ok, {PK, LK}};
+       {Need, Got} ->
+            {error, {bad_key_length, Need, Got}}
```
Comment (lukebakken) [inline src/riak_kv_pb_timeseries.erl]: This should be:

`{error, {bad_key_length, Got, Need}}`

File: src/riak_kv_pb_timeseries.erl (hunk 1 above)
Comment (alexmoore) [inline src/riak_kv_pb_timeseries.erl]: This ^^

File: src/riak_kv_pb_timeseries.erl [hunk 3]
```diff
@@ -52,18 +53,18 @@
...
+-define(E_FETCH,             1002).
+-define(E_IRREG,             1003).
+-define(E_PUT,               1004).
+-define(E_NOCREATE,          1005).
+-define(E_NOT_TS_TYPE,       1006).
+-define(E_MISSING_TYPE,      1007).
+-define(E_MISSING_TS_MODULE, 1008).
+-define(E_DELETE,            1009).
+-define(E_GET,               1010).
+-define(E_BAD_KEY_LENGTH,    1011).
```
Comment (lukebakken) [inline src/riak_kv_pb_timeseries.erl:66]: Awesome. I will update RIAK-1437. Thanks!

**Comment by lukebakken** [comment]
Tested with this Node.js client commit:

https://github.com/basho/riak-nodejs-client/commit/24dd080b05df5af08064eedfb08b3cb5ab46e3cb

**Comment by lukebakken** [comment]
+1 bc73dfa4a614cac22776bf5ee58ee2dae5c00522

**Comment by lukebakken** [comment]
+1 6674a28a2f1ac142711296ad35320ed9754d22b5

**Comment by lukebakken** [comment]
Ah I was wondering about that header and figured I just needed to update my local `riak_pb` checkout.

**Comment by lukebakken** [comment]
@borshop merge

**Comment by lukebakken** [comment]
Fixes #1256 (RIAK-2266)
